apps/judicial-system/backend/src/app/modules/case/guards/test/caseTypeGuard.spec.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  ExecutionContext,
+  ForbiddenException,
+  InternalServerErrorException,
+} from '@nestjs/common'
+
+import { CaseType } from '@island.is/judicial-system/types'
+
+import { randomListSplit } from '../../../../test'
+import { CaseTypeGuard } from '../caseType.guard'
+
+interface Then {
+  result: boolean
+  error: Error
+}
+
+type GivenWhenThen = () => Then
+
+describe('Case Type Guard', () => {
+  const mockRequest = jest.fn()
+  const [selectedTypes, remainingTypes] = randomListSplit(CaseType)
```

Comment:
Consider not using random values in tests. It does not matter what types are put here as long as they are not duplicated, so I am just going to approve.